### ISSUE TITLE
Exporting blocks needs to take variables with it

### ISSUE THREAD

**Issue by maryfries** [jmoenig/Snap--Build-Your-Own-Blocks#1133]
I'm exporting blocks from one project for use in another project, but some of the custom blocks use global variables (and quite a number of them!), and since I don't have them in this new project, I can't run the blocks.

Not only should all dependent _blocks_ be exported with any block export, BUT ALSO any dependent _variables_ need to go with it.

**Comment by jmoenig** [comment]
No, please don't  do this to the repo, and don't say things like *should* when things like scope are in the way! You are using the export blocks feature in ways that I did not want to support at all, and you are building blocks in ways that I also did not want to support at all. We should talk about this some place else.

**Comment by cycomachead** [comment]
Jens, I understand the intention behind the export mechanism, and also the desire to have pure and clean blocks, but I do think some dependency checking when exporting would be good. At the very least it would help to have useful error messages. There is some value, but really very little value!, in exporting a block which won't work when re-imported. There's so little value, that I'd almost be in favor of disallowing it.

For reference, this applies when exporting sprites too.
